Carbon/navigation/screens/Questionnaire/questionnaire.js

```diff
@@ -0,0 +1,447 @@
+        />
+        </View>
+        </>
+    )
+}
+
+function Diet({navigation}) {
+    const maxPoints = 10.0;
+    const [isDisabled,setIsDisabled] = useState(false);
+    const [pointPercent,setPointPercent] = useState(0);
+    const [buttonIndex, setButtonIndex] = useState(-1);
+    const changeIndex=(index)=>{
+        setButtonIndex(previousState=>index);
+    }
+    const disableButton=(points)=> {
+            setIsDisabled(previousState=>true);
+            setPointPercent(previousState=>points/maxPoints);
+    }
+
+    return (
+            <>
+            <View
+                style={{
+                    flex: 1,
+                    alignItems: 'center',
+                    justifyContent: 'center',
+                }}
+            >
+            <Text>What is your diet?</Text>
+            <View style={{
+                width:"100%",
+            }}
+            >
+            <Button
+                title="No Restrictions"
+                onPress={()=>{
+                disableButton(3);
+                setButtonIndex(0);
+                }}
+                color={buttonIndex==0 ? Colors.primary.RAISIN_BLACK: Colors.secondary.LIGHT_MINT}
+            />
+            <Button
+                title ="Pescatarian"
+                onPress={()=>{
+                disableButton(4);
+                setButtonIndex(1);
+                }}
+                color={buttonIndex==1 ? Colors.primary.RAISIN_BLACK: Colors.secondary.LIGHT_MINT}
+            />
+            <Button
+                title="Vegetarian"
+                onPress={()=>{
+                disableButton(6);
+                setButtonIndex(2);
+                }}
+                color={buttonIndex==2 ? Colors.primary.RAISIN_BLACK: Colors.secondary.LIGHT_MINT}
+            />
+            <Button
+                title ="Vegan/Plant-Based"
+                onPress={()=>{
+                disableButton(10);
+                setButtonIndex(3);
+                }}
+                color={buttonIndex==3 ? Colors.primary.RAISIN_BLACK: Colors.secondary.LIGHT_MINT}
+            />
+            <Text>{pointPercent}</Text>
+            </View>
+            </View>
+            <View style={{
+                justifyContent:'center',
+                flexDirection:"row",
+            }}>
+            <View style={{width:'50%'}}>
+            <Button
+            title="Return to Start"
+            color={Colors.primary.MINT}
+            onPress={() =>
+                navigation.goBack()
+            }
+            />
+            </View>
+            <View style={{width:'50%'}}>
+            <Button
+            title="Next Question"
+            color={Colors.primary.MINT}
+            onPress={() =>
+                navigation.navigate('q2',{dietScore:pointPercent})
+            }
+            disabled ={isDisabled ? false: true}
+            />
+            </View>
+            </View>
+            </>
+        )
+}
+
+function HouseholdPower({navigation,route}) {
```

Comment:
You should consider making these components instead of functions as well as giving them their own files to make the file structure cleaner.